lib/jobs/ansible-job.js

```diff
@@ -65,6 +86,24 @@ function ansibleJobFactory(
             return self._publishAnsibleResult(self.routingKey, data);
         })
         .then(function() {
+            try {
```

Comment:
I was able to rework the error handling and put the call to end the workflow in the callback of the file read.  (I just saw the part of your note regarding adding a promise - and although I had trouble with error handling, it seems to work OK now - let me know if you still think the promise should be in there).  Thx.